biogeochem/FatesAllometryMod.F90

```diff
@@ -1489,7 +1507,111 @@ subroutine dh2blmax_2pwr(d,p1,p2,dbh_maxh,c2b,blmax,dblmaxdd)
+     !    tropical forests. J. Geophys. Res.-Biogeosci. 125:
+     !    e2020JG005677. doi:10.1029/2020JG005677.
+     !
+     ! -----------------
+     !  Input arguments
+     ! -----------------
+     ! d        -- Diameter at breast height          [     cm]
+     ! h        -- Total tree height                  [      m]
+     ! dhdd     -- Height derivative with dbh         [   m/cm]
+     ! p1       -- Parameter 1 (log-intercept)        [     --]
+     ! p2       -- Parameter 2 (power, or log-slope)  [     --]
+     ! p3       -- Parameter 3 (power, or log-slope)  [     --]
+     ! slatop   -- Top-of-canopy specific leaf area   [  m2/gC]
+     ! dbh_maxh -- DBH at maximum height              [     cm]
+     ! c2b      -- Carbon to biomass multiplier ~ 2   [ kg/kgC]
+     !
+     ! ------------------
+     !  Output arguments
+     ! ------------------
+     ! blmax    -- Plant leaf biomass                 [    kgC]
+     ! dblmaxdd -- Plant leaf biomass derivative      [ kgC/cm]
+     !
+     !--------------------------------------------------------------------------
+
+
+     !--- Arguments
+     real(r8), intent(in)            :: d        ! plant diameter                 [    cm]
+     real(r8), intent(in)            :: h        ! plant height                   [     m]
+     real(r8), intent(in)            :: dhdd     ! Height derivative wrt diameter [  m/cm]
+     real(r8), intent(in)            :: p1       ! Log-intercept parameter        [     -]
+     real(r8), intent(in)            :: p2       ! Log-slope parameter for size   [     -]
+     real(r8), intent(in)            :: p3       ! Log-slope parameter for SLA    [     -]
+     real(r8), intent(in)            :: slatop   ! Top canopy specific leaf area  [ m2/gC]
+     real(r8), intent(in)            :: c2b      ! Carbon to biomass multiplier   [kg/kgC]
+     real(r8), intent(in)            :: dbh_maxh ! dbh at maximum height          [    cm]
+     real(r8), intent(out)           :: blmax    ! Leaf biomass                   [   kgC]
+     real(r8), intent(out), optional :: dblmaxdd ! Leaf biomass derivative        [kgC/cm]
+     !--- Local variables
+     real(r8) :: duse
+     !---~---
+
+
+
+     !--- Cap DBH
+     duse = min(d,dbh_maxh)
+     !---~---
+
+
+     !--- Find leaf biomass
+     blmax = p1 * (duse*duse*h)**p2 * slatop**p3 / c2b
+     !---~---
+
+
+     !---~---
+     !   Compute the leaf biomass derivative with DBH if needed.
+     !---~---
+     if (present(dblmaxdd)) then
+        if (d >= dbh_maxh) then
+           !---~---
+           !   Leaf area is capped at the maximum DBH. This may be removed in the
+           ! future.
+           !---~---
+           dblmaxdd = 0._r8
+           !---~---
+        else
+           !---~---
+           !   Find the leaf biomass derivative, noting that height is actually
+           ! a function of DBH.
+           !---~---
+           dblmaxdd = p2 * blmax * ( 2._r8 / duse + dhdd / h )
```

Comment:
@mpaiao does this need to swap the 2 for c2b?

Reply:
No, this 2 comes from the derivative of blmax, because of the DBH<sup>2</sup> term in the main allometric equation. I can add a note to indicate this.

Reply:
I opted for writing the derivative this way instead of the more explicit (like in `dh2bagw_chave2014`) because it reduces the number of operations (especially power operations). I could update the one in `dh2bagw_chave2014` too if it interests.

Reply:
i think its fine